### DISCUSSION TITLE
Incease maxPayloadLength upper limit and/or allow 0 (limitless)

### DISCUSSION THREAD

**Discussion by Zabrane** [uNetworking/uWebSockets.js#382] [Q&A] [answered]
My uWebSockets.js app daily handles large amount of data (`10-35GB` per conn) with no issue.
One of my colleagues tried to increase the `maxPayloadLength` to `100GB` for stress testing but it didn't work. Apparently, there's a `50GB` upper bound limit. 

```javascript
 maxPayloadLength: 100 * 1024 * 1024 * 1024 // failed
```

Is there any reason for that?

@alexhultman two suggestions if you don't mind:

1.  allow `maxPayloadLength` to take value `0` (no limit) and let the user handle the limit logic on his side 
2. increase the default `maxPayloadLength` to a much larger value: for example `100TB` (see [websocket-frame-size-limitation](https://stackoverflow.com/questions/20652194/websocket-frame-size-limitation/20658569#20658569))

**Comment by ghost** [comment]
If you want to transfer hundreds of GB you really shouldn't use one single WebSocket message. Use streaming HTTP and you can cut your memory usage in half at least?

**Comment by Zabrane** [comment]
@alexhultman what `maxPayloadLength` is for? Is it a limitation:
1. per message
2. per all messages inside 1 connection
3. for all ongoing connections

**Comment by hst-m** [comment, marked as answer]
Docs for [maxPayloadLength](https://unetworking.github.io/uWebSockets.js/generated/interfaces/websocketbehavior.html#maxpayloadlength) say per message

> Maximum length of received message. If a client tries to send you a message larger than this, the connection is immediately closed

**Comment by Zabrane** [comment]
Thanks for the pointer @hst-m

**Comment by Zabrane** [comment]
@alexhultman our service is `WebSocket` based. I'm afraid the Streaming won't help.

Anyway, i can close the issue. Thanks guys
